### ISSUE TITLE
E100 - uhd_usrp_probe timeout at readback test

### ISSUE THREAD

**Issue by repk** [balister/meta-ettus#4]
Hi,

Sorry for digging a 4 years old project. I am trying to compile linux kernel + uhd + gnuradio on ettus E100. Because sakoman repository is down I am forced to use the gumstix linux repository (with linux 3.5).

So I have checked out gumstix linux branch at the same SHA-1 that the one in  meta-ettus/e100-bsp/recipes-kernel/linux-overo/linux-overo_3.5.bb (i.e. 9d868d4024fe55192d0bace5ae5f2a84e918014b). Then I applied all patches in meta-ettus/e100-bsp/recipes-kernel/linux-overo/ettus-e1xx/ in correct order and used the defconfig + additionnal configs in there. Everything crossed compiled fine.

I then cross compiled an uhd_003_005_005 uploaded everything on the target and booted up fine.

But when I try to use uhd_usrp_probe I get a timeout with the following log : 
```
# modprobe usrp_e
# uhd_usrp_probe
linux; GNU C++ version 6.4.0; Boost_106600; UHD_003.005.005-0-unknown

-- Loading FPGA image: /usr/share/uhd/images/usrp_e100_fpga_v2.bin... done = 1
-- Initializing FPGA clock to 64.000000MHz...
-- USRP-E100 clock control: 10
--   r_counter: 2
--   a_counter: 0
--   b_counter: 20
--   prescaler: 8
--   vco_divider: 5
--   chan_divider: 5
--   vco_rate: 1600.000000MHz
--   chan_rate: 320.000000MHz
--   out_rate: 64.000000MHz
--
-- Opening device node /dev/usrp_e0...
-- Performing control readback test... Error: RuntimeError: fifo ctrl timed out looking for acks
```

I have taken a quick look at uhd code base and it seems to use gpio 147 as an interrupt to receive data from FPGA. It looks like this gpio is always on (even after flashing FPGA firmware). So uhd is constantly issuing USRP_E_READ_CTL32 ioctl to the driver that reads a successfully mapped gpmc requested region at (0x03000000). Here is an extract of IOMEM kernel mapping : 
```
# cat /proc/iomem
00000000-00000000 : omap2-nand.0
01000000-010000ff : smsc911x.0
  01000000-010000ff : smsc911x
02000000-020007ff : usrp_e
03000000-030007ff : usrp_e_c
48056000-48056fff : omap_dma_system.0
48060000-4806007f : omap_i2c.3
  48060000-4806007f : omap_i2c
48062000-48062fff : tll
[....]
```
I currently suspect that FPGA flashing process failed somehow, because GPIO147 is not even toggling after this step and keeps being in high state.

Do you have any idea about this issue, or any direction on where to look for ?

Thank you in advance.

**Comment by repk** [comment]
**tl;dr remove CONFIG_TOUCHSCREEN_ADS7846.**

The problem was that send_file_to_fpga() in lib/usrp/e100/fpga_downloader.cpp code from uhd tries to use a spidev in "/dev/spidev1.0". There are no error if opening this spidev node fails.

This node is missing from a 3.5 kernel it is supposed to be spi bus number 1 with CS as 0. As we can see in arch/arm/mach-omap2/board-overo.c, this device is only created when CONFIG_TOUCHSCREEN_ADS7846 or CONFIG_TOUCHSCREEN_ADS7846_MODULE is not defined. Removing CONFIG_TOUCHSCREEN_ADS7846 fixed the problem.

I could do a PR to remove CONFIG_TOUCHSCREEN_ADS7846 from the current defconfig if you want.
